mmv1/api/type.rb

```diff
@@ -517,7 +632,11 @@ def validate
         return if @__resource.nil? || @__resource.exclude || @exclude
 
         check :resource, type: ::String, required: true
-        check :imports, type: ::String, required: true
+        check :imports, type: ::String, required: TrueClass
+
+        # TODO: (camthornton) product reference may not exist yet
```

Comment:
This is because we're calling `validate` as part of loading the resource now, which doesn't guarantee it's part of a `::Product` yet?

Reply:
Yes, this validation will have to be called after the resources have been made a part of the Product